### PR TITLE
Respect the output flag when generating documentation

### PR DESCRIPTION
**Pull request by jpreese**
There was some odd behavior when constructing the relative path for the rego file with/without using the `--output` flag. This fixes that and adds some acceptance testing around documentation generation.

### CODE REVIEW

File: internal/commands/document.go [hunk 1]
```diff
@@ -38,26 +38,32 @@ func newDocCommand() *cobra.Command {
...
+	outputDirectory := filepath.Dir(viper.GetString("output"))
+	policyDocumentation, err := getPolicyDocumentation(path, outputDirectory)
 	if err != nil {
 		return fmt.Errorf("get policy documentation: %w", err)
 	}
 
-	err = ioutil.WriteFile(viper.GetString("output"), []byte(policyDocumentation), os.ModePerm)
+	outputPath := viper.GetString("output")
+	if outputPath == "" {
+		outputPath = "policies.md"
```
Comment (jalseth) [inline internal/commands/document.go:55]: Why are we using a conditional here rather than setting a default value in the flag?

File: internal/rego/rego.go [hunk 2]
```diff
@@ -124,9 +129,13 @@ func loadRegoFiles(files []string) ([]File, error) {
 	return regoFiles, nil
 }
 
-func getModuleRulesActions(module *ast.Module) []string {
+func getModuleRulesActions(module *ast.Module) ([]string, error) {
+	re, err := regexp.Compile(`^\s*([a-z]+)\s*\[\s*msg`)
+	if err != nil {
+		return nil, fmt.Errorf("parse rule: %w", err)
```
Comment (jalseth) [inline internal/rego/rego.go:135]: Error message isn't quite right. It's about compiling a regex, not parsing the rego rules

File: internal/commands/document.go (hunk 1 above)
Comment (jpreese) [reply internal/commands/document.go:55]: I had a reason for this initially... but I don't recall why. Shouldn't be needed. I'll fix this.

File: internal/rego/rego.go (hunk 2 above)
Comment (jpreese) [reply internal/rego/rego.go:135]: Yeah, that's fair. I'll update the message.

File: internal/commands/document.go (hunk 1 above)
Comment (jpreese) [reply internal/commands/document.go]: Fixed!

File: internal/rego/rego.go (hunk 2 above)
Comment (jpreese) [reply internal/rego/rego.go]: All set!

**Comment by jpreese** [comment]
@jalseth comments addressed and pushed.

File: internal/commands/document.go [hunk 3]
```diff
@@ -79,21 +73,19 @@ func getPolicyDocumentation(path string, outputDirectory string) (string, error)
...
-		ruleTypes := strings.Join(policy.RulesActions, ", ")
-
 		for _, policyCommentBlock := range policyCommentBlocks {
+			relPath, err := filepath.Rel(outputDirectory, policy.FilePath)
+			if err != nil {
+				return "", fmt.Errorf("rel path: %w", err)
+			}
+
+			relDir := filepath.Dir(relPath)
+			ruleTypes := strings.Join(policy.RulesActions, ", ")
```
Comment (jalseth) [inline internal/commands/document.go:88]: Why was this moved inside the policyCommentBlocks iterator? This won't change until the next policy so we're potentially computing these more often than needed.

Comment (jpreese) [reply internal/commands/document.go:88]: It was mostly a readability play. I agree it's excess compute, but it feels wrong to keep them so far away from their use. I'm not strongly opinionated about it either way. This is probably a smell that this construction needs some refactoring love so things aren't so far apart from one another.

Comment (jalseth) [reply internal/commands/document.go:93]: I think this opens the larger discussion of should we allow/support more than one comment block per policy? Our tooling is really built around having a bunch of small policy files that check for one thing, even if that may be split out into multiple rules. If we didn't have to iterate over multiple policy blocks, that would resolve this as well.

Comment (jpreese) [reply internal/commands/document.go:93]: I'm more inclined to leans towards the idea of one comment block per policy file. It seems a little silly to support different `Kinds` for different `violations` given the same policy file. Smells of needing to split the file up. So if we only supported a single set of Kind matchers in a file, you could argue that means a single comment block as well.

The only counterexample I can think of at the moment is if you wanted to be able to keep the comment block with the violation itself--and the documentation would then be summing up all of those blocks:

```rego
// All containers must do stuff
// Here is an example of a valid container
// ..yaml..
// Yeah that's pretty valid, amirite?
// Kinds @v1/Pod

// Containers must set a memory limit.
// (you'd probably change this to have it make way more sense as a cohesive document)
violation[] {}

// Containers must set a request limit.
violation[] {}
```

doc would be (something like)

```markdown
All containers must do stuff
Here is an example of a valid container
..yaml..
Yeah that's pretty valid, amirite?

Containers must set a memory limit.

Containers must set a request limit.
```

All that said, it's more of something to consider as something that might pop up--but at least for now I'd rather optimize for the case of a single block.

Comment (jalseth) [reply internal/commands/document.go:93]: Agreed. I think we can merge this as it is and open a separate PR for addressing this issue.

**Comment by jalseth** [comment]
LGTM!
